src/main/java/dk/kb/present/holdback/PurposeMatrixSheet.java

```diff
@@ -0,0 +1,134 @@
+        }
+
+        for (Row row : purposeMatrixSheet) {
+            Cell indholdFraCell = row.getCell(1);
+            Cell indholdTilCell = row.getCell(2);
+
+            // Get all IndholdFra values.
+            if (indholdFraCell != null && indholdFraCell.getCellType() == CellType.NUMERIC) {
+                int indholdFraValue = (int) indholdFraCell.getNumericCellValue();
+                indholdFra.add(indholdFraValue);
+            }
+
+            //Get all IndholdTil values.
+            if (indholdTilCell != null && indholdTilCell.getCellType() == CellType.NUMERIC) {
+                int indholdTilValue = (int) indholdTilCell.getNumericCellValue();
+                indholdTil.add(indholdTilValue);
+            }
+
+            //Create a map entry for each column
+            Iterator<Cell> cellIterator = row.cellIterator();
+            if (row.getRowNum() == 0){
+                while (cellIterator.hasNext()){
+                    Cell formNrNameCell = cellIterator.next();
+
+                    if (formNrNameCell.getCellType() == CellType.STRING &&
+                        formNrNameCell.getStringCellValue().startsWith("Form")){
+                        formNr.add(formNrNameCell.getStringCellValue());
+                    }
+                }
+            } else {
+                // Iterate over each row in the sheet
+                for (int colIndex = 3; colIndex < NUMBER_OF_COLUMNS; colIndex++) {
+                    Cell cell = row.getCell(colIndex);
+                    if (cell.getCellType() == CellType.NUMERIC){
+                        formNrValues.get(colIndex).add(String.valueOf( (int) cell.getNumericCellValue()));
+                    } else {
+                        formNrValues.get(colIndex).add(cell.getStringCellValue());
+                    }
+                }
+            }
+        }
+
+        // Remove the first item 3 times to target empty columns for Id, IndholdFra and IndholdTil
+        formNrValues.subList(0, 3).clear();
+
+        if (formNrValues.size() != formNr.size()){
+            throw new InternalServiceException("There is a difference between amount of formNrs and formNrValues. Their respective sizes are: " + formNr.size() + " and " + formNrValues.size());
+        }
+
+        if (indholdTil.size() != indholdFra.size()){
+            throw new InternalServiceException("There is a difference between the sizes of the lists 'indholdTil' and 'indholdFra'. Their respective sizes are: " + indholdTil.size() + " and " + indholdFra.size());
+        }
+
+        for (List<String> list : formNrValues) {
+            if (list.size() != indholdFra.size()){
+                throw new InternalServiceException("There is a difference between the sizes of a list in the nested list 'formNrValues' and 'indholdFra'. Their respective sizes are: " + list.size() + " and " + indholdFra.size());
+            }
+        }
+
+        log.info("Initialized PurposeMatrixSheet with the following values: \n" +
```

Comment:
This should be moved to the toString() method  for the class. This is not just a log statement, but a representation of the class.

 And here just call toString() to log it.
Maybe same situation for the other sheet classes.